public/app/features/search/components/SearchWrapper.tsx

```diff
@@ -1,34 +1,48 @@
-import React, { FC, useState, useEffect } from 'react';
-import { appEvents } from 'app/core/core';
-import { CoreEvents } from 'app/types';
+import React, { FC, memo } from 'react';
+import { MapDispatchToProps, MapStateToProps } from 'react-redux';
+import { getLocationQuery } from 'app/core/selectors/location';
+import { updateLocation } from 'app/core/reducers/location';
+import { connectWithStore } from 'app/core/utils/connectWithReduxStore';
+import { StoreState } from 'app/types';
 import { DashboardSearch } from './DashboardSearch';
-import { OpenSearchParams } from '../types';
-
-export const SearchWrapper: FC = () => {
-  const [isOpen, setIsOpen] = useState(false);
-  const [payload, setPayload] = useState({});
-
-  useEffect(() => {
-    const openSearch = (payload: OpenSearchParams) => {
-      setIsOpen(true);
-      setPayload(payload);
-    };
-
-    const closeOnItemClick = (payload: any) => {
-      // Detect if the event was emitted by clicking on search item
-      if (payload?.target === 'search-item' && isOpen) {
-        setIsOpen(false);
-      }
-    };
-
-    appEvents.on(CoreEvents.showDashSearch, openSearch);
-    appEvents.on(CoreEvents.hideDashSearch, closeOnItemClick);
-
-    return () => {
-      appEvents.off(CoreEvents.showDashSearch, openSearch);
-      appEvents.off(CoreEvents.hideDashSearch, closeOnItemClick);
-    };
-  }, [isOpen]);
-
-  return isOpen ? <DashboardSearch onCloseSearch={() => setIsOpen(false)} payload={payload} /> : null;
+
+interface OwnProps {
+  search?: string | null;
+  folder?: string;
+  queryText?: string;
+  filter?: string;
+}
+
+interface DispatchProps {
+  updateLocation: typeof updateLocation;
+}
+
+export type Props = OwnProps & DispatchProps;
+
+export const SearchWrapper: FC<Props> = memo(({ search, folder, updateLocation }) => {
+  const isOpen = search === 'open';
+
+  const closeSearch = () => {
+    if (search === 'open') {
+      updateLocation({
+        query: {
+          search: null,
```

Comment:
Probably this should fix a bug
```suggestion
          search: null,
          folder: null,
```

Reply:
Yeah, good catch, forgot to reset the folder :)